### PR TITLE
Add opening as secondary

### PR DESCRIPTION
**Pull request by oxade**
Allows opening a DB in secondary mode

### CODE REVIEW

File: crates/typed-store/src/rocks/mod.rs [hunk 1]
```diff
@@ -474,6 +480,63 @@ pub fn open_cf_opts<P: AsRef<Path>>(
...
+    secondary_path: Option<P>,
+    db_options: Option<rocksdb::Options>,
+    opt_cfs: &[(&str, &rocksdb::Options)],
+) -> Result<Arc<rocksdb::DBWithThreadMode<MultiThreaded>>, TypedStoreError> {
+    // Customize database options
+    let mut options = db_options.unwrap_or_default();
+
+    // One common issue when running tests on Mac is that the default ulimit is too low,
+    // leading to I/O errors such as "Too many open files". Raising fdlimit to bypass it.
+    options.set_max_open_files(-1);
```
Comment (huitseeker) [inline crates/typed-store/src/rocks/mod.rs]: We use a one-liner from [`fdlimit`](https://github.com/paritytech/fdlimit) for this in the Sui code base, and could use it here.

File: crates/typed-store/src/rocks/mod.rs [hunk 2]
```diff
@@ -474,6 +480,63 @@ pub fn open_cf_opts<P: AsRef<Path>>(
...
+    }
+
+    let primary_path = primary_path.as_ref().to_path_buf();
+    let secondary_path = secondary_path
+        .map(|q| q.as_ref().to_path_buf())
+        .unwrap_or_else(|| {
+            let mut s = primary_path.clone();
+            s.pop();
+            s.push("SECONDARY");
+            s.as_path().to_path_buf()
```
Comment (huitseeker) [inline crates/typed-store/src/rocks/mod.rs:517]: You could use the [`std::env!` macro](https://doc.rust-lang.org/std/macro.env.html) to allow overriding this as a one-liner.

**Review by huitseeker — COMMENTED** [review]
Tip top, I made just a couple suggestions.

File: crates/typed-store/src/rocks/mod.rs (hunk 2 above)
Comment (oxade) [reply crates/typed-store/src/rocks/mod.rs:517]: That'll cause override at compile time. Why would we want that?

File: crates/typed-store/src/rocks/mod.rs (hunk 1 above)
Comment (oxade) [reply crates/typed-store/src/rocks/mod.rs]: Got it. Will address.

File: crates/typed-store/src/rocks/mod.rs (hunk 2 above)
Comment (huitseeker) [reply crates/typed-store/src/rocks/mod.rs:517]: 🤦 compile time overrides, not the best idea.
I meant trying to probe the value at runtime. Or a launch argument. Or something other configuration that allows not using the hard coded string.

**Review by huitseeker — APPROVED** [review]
Thanks! Nice work, apologies I had missed the `secondary_path` argument.

**Pull request merged**